nncf/ptq/api/quantization.py

```diff
@@ -0,0 +1,113 @@
+"""
+ Copyright (c) 2022 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from typing import Callable
+from typing import Optional
+
+from nncf.ptq.api.types import ModelType
+from nncf.ptq.backend import determine_backend
+from nncf.ptq.backend import BackendType
+from nncf.ptq.data.dataloader import NNCFDataLoader
+
+
+def quantize(model: ModelType,
+             calibration_dataset: NNCFDataLoader,
+             preset: str = 'performance',
```

Comment:
All the string parameters here are enums really. Can we change them to enums? This can be more intuitive for the user to select the proper configuration.
BTW, NNCF should have preset already, the same as POT has.